scripts/utils/trading_strategy_helpers.js

```diff
@@ -85,37 +116,60 @@ const isOnlySafeOwner = async function(masterAddress, ownedAddress, artifacts) {
 
 /**
  * Queries EVM for ERC20 token details by address
- * and returns a list of detailed token information.
- * @param {SmartContract} exchange BatchExchange, contract, or any contract implementing `tokenIdToAddressMap`
- * @param {integer[]} tokenIds list of token ids whose data is to be fetch from EVM
- * @return {TokenObject[]} list of detailed/relevant token information
+ * and returns a list of promises of detailed token information.
+ * @param {Address[]} tokenAddresses list of *unique* token addresses whose data is to be fetch from the EVM
+ * @return {Promise<TokenObject>[]} list of detailed/relevant token information
  */
-const globalTokenObjects = {}
-const fetchTokenInfo = async function(exchange, tokenIds, artifacts, debug = false) {
+const globalTokenPromisesFromAddress = {}
```

Comment:
I think the doc comment is for the constant and not the function how it is currently.